modules/navier_stokes/src/auxkernels/NSMachAux.C

```diff
@@ -0,0 +1,50 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "NSMachAux.h"
+#include "MooseMesh.h"
+
+template<>
+InputParameters validParams<NSMachAux>()
+{
+  InputParameters params = validParams<AuxKernel>();
+
+  // Mark variables as required
```

Comment:
Some are required, some are not. We can see which ones from the name. Also, for 1D run, `y-velocity` should not be required.